src/main/java/core/basesyntax/ArrayList.java

```diff
@@ -1,48 +1,139 @@
 package core.basesyntax;
 
+import java.util.NoSuchElementException;
+
 public class ArrayList<T> implements List<T> {
+    private T[] arrayList;
+    private int size = 0;
+    private int emptyPlaces = 10;
+    private final double ARRAY_ENLARGEMENT_PERCENTAGE = 0.5;
+
+    public ArrayList() {
+        arrayList = (T[]) new Object[emptyPlaces];
+    }
+
+    public void enlarge() {
+        emptyPlaces = (int) (size * ARRAY_ENLARGEMENT_PERCENTAGE);
+        T[] temporaryEnlargedArrayList = (T[]) new Object[size + emptyPlaces];
```

Comment:
Use System.arraycopy() to copy elements from one array to another instead of a manual loop to improve performance and readability.